packages/teams-js/test/public/marketplace.spec.ts

```diff
@@ -0,0 +1,476 @@
+          );
+        });
+
+        Object.values(FrameContexts).forEach((context) => {
+          if (context === FrameContexts.content || context === FrameContexts.task) {
+            it(`marketplace.addOrUpdateCartItems should throw error when marketplace is not supported when initialized with ${context} context`, async () => {
+              await utils.initializeWithContext(context);
+              utils.setRuntimeConfig({ apiVersion: 2, supports: {} });
+              expect(marketplace.addOrUpdateCartItems(cartItems)).rejects.toEqual(errorNotSupportedOnPlatform);
+            });
+
+            it('marketplace.addOrUpdateCartItems should successfully send the addOrUpdateCartItems message', async () => {
+              await utils.initializeWithContext(context);
+              utils.setRuntimeConfig({ apiVersion: 2, supports: { marketplace: {} } });
+
+              const promise = marketplace.addOrUpdateCartItems(cartItems);
+
+              const addOrUpdateCartItemsMessage = utils.findMessageByFunc('marketplace.addOrUpdateCartItems');
+              validateExpectedArgumentsInRequest(
+                addOrUpdateCartItemsMessage,
+                'marketplace.addOrUpdateCartItems',
+                MatcherType.ToStrictEqual,
+                cartItems,
+              );
+
+              utils.respondToMessage(addOrUpdateCartItemsMessage!);
+              await promise;
+            });
+          } else {
+            it(`marketplace.addOrUpdateCartItems should not allow calls from ${context} context`, async () => {
+              await utils.initializeWithContext(context);
+              await expect(marketplace.addOrUpdateCartItems(cartItems)).rejects.toThrowError(
+                `This call is only allowed in following contexts: ${JSON.stringify([
+                  FrameContexts.content,
+                  FrameContexts.task,
+                ])}. Current context: "${context}".`,
+              );
+            });
+          }
+        });
+      });
+
+      describe('Testing marketplace.removeCartItems function', () => {
+        const cartItemIds = ['001', '002', '003'];
+
+        it('marketplace.removeCartItems should not allow calls before initialization', async () => {
+          await expect(marketplace.removeCartItems(cartItemIds)).rejects.toThrowError(
+            new Error(errorLibraryNotInitialized),
+          );
+        });
+
+        it('marketplace.removeCartItems should throw error with empty array input', async () => {
+          await utils.initializeWithContext(FrameContexts.content);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { marketplace: {} } });
+          expect(marketplace.removeCartItems([])).rejects.toThrowError(
+            new Error('cartItemIds must be a non-empty array'),
+          );
+        });
+
+        Object.values(FrameContexts).forEach((context) => {
+          if (context === FrameContexts.content || context === FrameContexts.task) {
```

Comment:
should have unit tests that test the "not an array" and "empty array" cases

Reply:
(The test case you have above only tests empty array, and it should b e moved into this forEach so we test it on all valid frameContexts)